Disable image verification by default

### What is wrong:

After building example image, I tried to install it with command
`elemental --debug install -d ttl.sh/example-mjura:8h /dev/vdb`

as results I have got following stacktrace

```
INFO[2022-05-04T13:59:21Z] Creating file system image /run/cos/state/cOS/active.img 
DEBU[2022-05-04T13:59:21Z] Running cmd: 'mkfs.ext2 -L COS_ACTIVE /run/cos/state/cOS/active.img' 
DEBU[2022-05-04T13:59:21Z] Mounting image COS_ACTIVE                    
DEBU[2022-05-04T13:59:21Z] Running cmd: 'losetup --show -f /run/cos/state/cOS/active.img' 
INFO[2022-05-04T13:59:22Z] Copying COS_ACTIVE image...                  
INFO[2022-05-04T13:59:22Z] Unpacking docker image: ttl.sh/example-mjura:8h 
  ERROR     Plugin luet-mtree at /usr/bin/luet-mtree had an error: error while executing plugin: exit status 1
panic: fatal error

goroutine 104 [running]:
github.com/mudler/luet/pkg/api/core/logger.(*Logger).Fatal(0x40c527, {0xc0318229d0, 0x11801e0, 0xc0014fa501})
	/home/mjura/go/pkg/mod/github.com/mudler/luet@v0.0.0-20220127164253-6f77fa2b3a72/pkg/api/core/logger/logger.go:255 +0x47
github.com/mudler/luet/pkg/api/core/bus.(*Bus).Initialize.func1(0xc000262880, 0xc031569380)
	/home/mjura/go/pkg/mod/github.com/mudler/luet@v0.0.0-20220127164253-6f77fa2b3a72/pkg/api/core/bus/events.go:108 +0x45d
reflect.Value.call({0x118f6e0, 0xc0000d09f0, 0x13}, {0x13829cb, 0x4}, {0xc0315695f0, 0x2, 0x2})
	/usr/lib64/go/1.17/src/reflect/value.go:556 +0x845
reflect.Value.Call({0x118f6e0, 0xc0000d09f0, 0x41ab00}, {0xc0315695f0, 0x2, 0x2})
	/usr/lib64/go/1.17/src/reflect/value.go:339 +0xc5
github.com/chuckpreslar/emission.(*Emitter).Emit.func1({0x118f6e0, 0xc0000d09f0, 0x0})
	/home/mjura/go/pkg/mod/github.com/chuckpreslar/emission@v0.0.0-20170206194824-a7ddd980baf9/emitter.go:209 +0x439
created by github.com/chuckpreslar/emission.(*Emitter).Emit
	/home/mjura/go/pkg/mod/github.com/chuckpreslar/emission@v0.0.0-20170206194824-a7ddd980baf9/emitter.go:184 +0x2d6
```

### Solution:

To resolve this problem I had to disable image verification

`elemental --debug install --no-verify -d ttl.sh/example-mjura:8h /dev/vdb`

### What should be done:

We should probably disable verification by default, or at least improve the error message, but if you didn't pushed any hash for your image it fails, as tries to check against a mtree

### PR's:

- [ ] https://github.com/rancher-sandbox/elemental/pull/208
- [ ] https://github.com/rancher-sandbox/ros-installer/pull/2
- [ ] https://github.com/mjura/cOS-toolkit/commit/dd12baccc7cf6bae6f3d918793ff0026f4e8a7fd

--- a/cmd/flags.go
+++ b/cmd/flags.go
@@ -54,7 +54,7 @@ func addResetFlags(cmd *cobra.Command) {
 	_ = cmd.Flags().MarkDeprecated("docker-image", "'docker-image' is deprecated please use 'system' instead")
 
 	cmd.Flags().String("system.uri", "", "Sets the system image source and its type (e.g. 'docker:registry.org/image:tag')")
-	cmd.Flags().Bool("no-verify", false, "Disable mtree checksum verification (requires images manifests generated with mtree separately)")
+	cmd.Flags().Bool("verify", false, "Enable mtree checksum verification (requires images manifests generated with mtree separately)")
 	cmd.Flags().Bool("strict", false, "Enable strict check of hooks (They need to exit with 0)")
 
 	addCosignFlags(cmd)

--- a/docs/elemental_install.md
+++ b/docs/elemental_install.md
@@ -19,7 +19,7 @@ elemental install DEVICE [flags]
   -i, --iso string                       Performs an installation from the ISO url
       --local                            Use an image from local cache
       --no-format                        Don’t format disks. It is implied that COS_STATE, COS_RECOVERY, COS_PERSISTENT, COS_OEM are already existing
-      --no-verify                        Disable mtree checksum verification (requires images manifests generated with mtree separately)
+      --verify                           Enable mtree checksum verification (requires images manifests generated with mtree separately)
       --part-table string                Partition table type to use (default "gpt")
       --poweroff                         Shutdown the system after install
       --reboot                           Reboot the system after install

--- a/docs/elemental_reset.md
+++ b/docs/elemental_reset.md
@@ -12,7 +12,7 @@ elemental reset [flags]
       --cosign              Enable cosign verification (requires images with signatures)
       --cosign-key string   Sets the URL of the public key to be used by cosign validation
   -h, --help                help for reset
-      --no-verify           Disable mtree checksum verification (requires images manifests generated with mtree separately)
+      --verify              Enable mtree checksum verification (requires images manifests generated with mtree separately)
       --poweroff            Shutdown the system after install
       --reboot              Reboot the system after install
       --reset-persistent    Clear persistent partitions

--- a/docs/elemental_upgrade.md
+++ b/docs/elemental_upgrade.md
@@ -13,7 +13,7 @@ elemental upgrade [flags]
       --cosign-key string                Sets the URL of the public key to be used by cosign validation
   -h, --help                             help for upgrade
       --local                            Use an image from local cache
-      --no-verify                        Disable mtree checksum verification (requires images manifests generated with mtree separately)
+      --verify                           Enable mtree checksum verification (requires images manifests generated with mtree separately)
       --poweroff                         Shutdown the system after install
       --reboot                           Reboot the system after install
       --recovery                         Upgrade the recovery

--- a/pkg/config/config.go
+++ b/pkg/config/config.go
@@ -163,7 +163,7 @@ func NewRunConfig(opts ...GenericOptions) *v1.RunConfig {
 // mapstructure unmarshal already took place.
 func CoOccurrenceConfig(cfg *v1.Config) {
 	// Set Luet plugins, we only use the mtree plugin for now
-	if !cfg.NoVerify {
+	if cfg.Verify {
 		cfg.Luet.SetPlugins(constants.LuetMtreePlugin)
 	}
 }

--- a/pkg/types/v1/config.go
+++ b/pkg/types/v1/config.go
@@ -46,7 +46,7 @@ type Config struct {
 	Luet                      LuetInterface
 	Client                    HTTPClient
 	Cosign                    bool         `yaml:"cosign,omitempty" mapstructure:"cosign"`
-	NoVerify                  bool         `yaml:"no-verify,omitempty" mapstructure:"no-verify"`
+	Verify                    bool         `yaml:"verify,omitempty" mapstructure:"verify"`
 	CosignPubKey              string       `yaml:"cosign-key,omitempty" mapstructure:"cosign-key"`
 	LocalImage                bool         `yaml:"local,omitempty" mapstructure:"local"`
 	Repos                     []Repository `yaml:"repositories,omitempty" mapstructure:"repositories"`
@@ -58,7 +58,7 @@ type Config struct {
 // if unsolvable inconsistencies are found
 func (c *Config) Sanitize() error {
 	// Set Luet plugins, we only use the mtree plugin for now
-	if !c.NoVerify {
+	if c.Verify {
 		c.Luet.SetPlugins(constants.LuetMtreePlugin)
 	}
 	return nil

--- a/pkg/types/v1/config_test.go
+++ b/pkg/types/v1/config_test.go
@@ -180,6 +180,7 @@ var _ = Describe("Types", Label("types", "config"), func() {
 	Describe("RunConfig", func() {
 		It("runs sanitize method", func() {
 			cfg := config.NewRunConfig(config.WithMounter(v1mocks.NewErrorMounter()))
+			cfg.Config.Verify = true
 
 			// Sets the luet mtree pluing
 			err := cfg.Sanitize()
@@ -190,6 +191,7 @@ var _ = Describe("Types", Label("types", "config"), func() {
 	Describe("BuildConfig", func() {
 		It("runs sanitize method", func() {
 			cfg := config.NewBuildConfig(config.WithMounter(v1mocks.NewErrorMounter()))
+			cfg.Config.Verify = true
 
 			// Sets the luet mtree pluing
 			err := cfg.Sanitize()

